src/fluree/db/session.cljc

```diff
@@ -79,71 +65,150 @@
-                                                {:status 400
-                                                 :error  :db/unavailable})))
-              last-indexed-db (<? (storage/reify-db conn network dbid blank-db (:index db-info)))
-              latest-block    (:block db-info)
-              db              (when last-indexed-db
-                                (loop [db         last-indexed-db
-                                       next-block (-> last-indexed-db :block inc)]
-                                  (if (> next-block latest-block)
-                                    db
-                                    (let [block-data (<? (storage/read-block conn network dbid next-block))]
-                                      (if block-data
-                                        (let [{:keys [flakes block t]} block-data
-                                              db* (<? (dbproto/-with db block flakes))]
-                                          (recur db* (inc next-block)))
-                                        (throw (ex-info (str "Error reading block " next-block " for db: " network "/" dbid ".")
-                                                        {:status 500 :error :db/unexpected-error})))))))
-              db-schema       (schema/schema-map db)        ;; returns async chan
-              db-settings     (schema/setting-map db)       ;; returns async chan
-              db*             (assoc db :schema (<? db-schema)
-                                        :settings (<? db-settings))]
-          (async/put! pc db*))
-        (catch* e
-                (async/put! pc e))))
-    pc))
+(defn ready?
+  [db-info]
+  (= :ready
+     (-> db-info :status keyword)))
+
+(defn load-ledger-info
+  [conn network dbid]
+  (go-try
+   (let [ledger-info (<? (ops/ledger-info-async conn [network dbid]))]
+     (if (empty? ledger-info)
+       (throw (ex-info (str "Ledger " network "/" dbid
+                            " is not found on this ledger group.")
+                       {:status 400
+                        :error  :db/unavailable}))
+       (if-not (ready? ledger-info)
+         (throw (ex-info (str "Ledger " network "/" dbid
+                              " is not currently available. Status is: "
+                              (:status ledger-info) ".")
+                         {:status 400
+                          :error  :db/unavailable}))
+         ledger-info)))))
+
+(defn load-current-db
+  [conn {:keys [network dbid] :as blank-db}]
+  (go-try
+   (let [{:keys [index], latest-block :block, :as ledger-info}
+         (<? (load-ledger-info conn network dbid))]
+     (when-let [indexed-db (<? (storage/reify-db conn network dbid blank-db index))]
+       (loop [db         indexed-db
+              next-block (-> indexed-db :block inc)]
+         (if (> next-block latest-block)
+           (let [schema   (<? (schema/schema-map db))
+                 settings (<? (schema/setting-map db))]
```

Comment:
setting and schema loads were done in parallel previously - each requires an index-range call. I don't know the realistic time savings, but I think probably OK like this.